skfda/preprocessing/feature_construction/_function_transformers.py

```diff
@@ -0,0 +1,196 @@
+"""Function transformers for feature construction techniques."""
+from typing import Optional, Sequence, Tuple
+
+from sklearn.base import BaseEstimator
+
+from ..._utils import TransformerMixin
+from ...exploratory.stats._functional_transformers import (
+    local_averages,
+    number_up_crossings,
+    occupation_measure,
+)
+from ...representation._typing import NDArrayFloat, Union
+from ...representation.basis import FDataBasis
+from ...representation.grid import FDataGrid
+
+
+class LocalAveragesTransformer(BaseEstimator, TransformerMixin):
+    """
+    Transformer that works as an adapter for the local_averages function.
+
+    Args:
+        n_intervals: number of intervals we want to consider.
+
+    Example:
+        We import the Berkeley Growth Study dataset.
+        We will use only the first 3 samples to make the
+        example easy.
+        >>> from skfda.datasets import fetch_growth
+        >>> dataset = fetch_growth(return_X_y=True)[0]
+        >>> X = dataset[:3]
+
+        Then we decide how many intervals we want to consider (in our case 2)
+        and call the function with the dataset.
+        >>> import numpy as np
+        >>> from skfda.preprocessing.feature_construction import (
+        ...     LocalAveragesTransformer,
+        ... )
+        >>> local_averages = LocalAveragesTransformer(2)
+        >>> np.around(local_averages.fit_transform(X), decimals=2)
+        array([[[ 116.94],
```

Comment:
This should be a 2D matrix to be able to use it in further scikit-learn estimators.